inflight_batcher_llm/src/libtensorrtllm.cc

```diff
@@ -605,17 +605,22 @@ class WorkItemsQueue
         mPendingWorkItems.pop_front();
         mPendingWorkItemsReqIds.erase(workItem->requestId());
 
-        bool markedInProgress;
         // Check if work item has been stopped
-        if (mStoppedReqIds.find(workItem->requestId()) == mStoppedReqIds.end())
+        bool is_stopped = mStoppedReqIds.find(workItem->requestId()) != mStoppedReqIds.end();
+
+        // Check if the Triton request behind is cancelled
+        bool is_cancelled = false;
+        TRITONBACKEND_ResponseFactoryIsCancelled(workItem->response_factory(), &is_cancelled);
+
+        bool markedInProgress = false;
+        if (!is_stopped && !is_cancelled)
         {
             mInProgressWorkItems.emplace(std::make_pair(workItem->requestId(), workItem));
             markedInProgress = true;
         }
-        else
+        else if (is_stopped)
```

Comment:
Up to preference, but you can also leave this as is safe to erase a non-existent key.

Reply:
Removing the ` if (is_stopped)` looks cleaner. Updated.